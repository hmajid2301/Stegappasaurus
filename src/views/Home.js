import { Icon } from 'native-base';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';

import CustomHeader from '~/components/CustomHeader';

import CustomTabNavigator from './Home/CustomTabNavigator';
import styles from './Home/styles';


class Home extends Component {
  static navigationOptions = {
    drawerLabel: 'Home',
    drawerIcon: ({ tintColor }) => (
      <Icon name='home' type='FontAwesome' style={{ color: tintColor }}/>
    ),
  };

  static propTypes = {
    navigation: PropTypes.object.isRequired,
    screenProps: PropTypes.object.isRequired,
    primaryColor: PropTypes.string.isRequired,
  };

  static router = CustomTabNavigator.router;

  render() {
    const { theme } = this.props.screenProps;

    return (
      <View style={styles.container}>
        <CustomHeader
          color={this.props.primaryColor}
          navigation={this.props.navigation}
          theme={theme}
        />
        <CustomTabNavigator
          navigation={this.props.navigation}
          screenProps={{ theme }}
        />
      </View>
    );
  }
}

const mapStateToProps = state => ({
  primaryColor: state.TogglePrimaryColor.colorData.color,
});

export default connect(mapStateToProps, null)(Home);
