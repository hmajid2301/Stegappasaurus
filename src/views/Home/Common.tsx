import React from 'react';
import {Dimensions, TextStyle, View} from 'react-native';
import {Icon} from 'react-native-elements';
import styled from 'styled-components/native';

import {Actions} from '~/actions/Steganography';
import {pureWhite} from '~/constants/colors';
import {body} from '~/constants/fonts';
import {TabColors, ThemeColors} from '~/constants/types';

const pageWidth = Dimensions.get('window').width;

export const getInnerProgressComponent = (
  action: Actions,
  progress: number,
  isEncoding = true,
) => {
  let component = <View />;

  if (action === 'encoding' || action === 'decoding') {
    component = <ProgressText>{Math.ceil(progress)}%</ProgressText>;
  } else if (action === 'done') {
    component = (
      <View>
        {isEncoding || (
          <Icon color={pureWhite} name="share" size={130} type="font-awesome" />
        )}
        <ProgressText>{Math.ceil(progress)}%</ProgressText>
      </View>
    );
  } else if (action === 'getting_image_data') {
    component = <ProgressText>Getting Image Data</ProgressText>;
  } else if (action === 'setting_image_data') {
    component = (
      <View>
        <ProgressText>Saving New Image</ProgressText>
      </View>
    );
  } else {
    component = (
      <View>
        <ProgressText>Starting</ProgressText>
      </View>
    );
  }

  return component;
};

const ProgressText = styled.Text`
  color: black;
  font-family: ${body};
  font-size: 30;
`;

export const MainContainer = styled.View<{background: ThemeColors}>`
  flex: 1;
  background-color: ${props => props.background};
`;

export const ButtonsContainer = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: center;
  margin-right: 2;
  margin-top: 2;
`;

export const TouchableButton = styled.TouchableOpacity<{button: TabColors}>`
  background-color: ${props => props.button};
  flex: 1;
  height: ${pageWidth / 3};
  justify-content: center;
  margin-left: 2;
  margin-top: 2;
`;

export const PhotoAlbumContainer = styled.View`
  flex: 1;
  margin-right: 2;
`;

export const iconStyle: TextStyle = {
  color: pureWhite,
  fontSize: 40,
  textAlign: 'center',
};
