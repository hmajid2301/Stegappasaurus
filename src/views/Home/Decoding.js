import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { connect } from 'react-redux';

import { togglePrimaryColor } from '~/redux/actions';
import { PRIMARY_COLORS } from '~/util/constants';

import styles from './Decoding/styles';


class Decoding extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
    screenProps: PropTypes.object.isRequired,
    togglePrimaryColor: PropTypes.func.isRequired,
  };

  componentDidMount() {
    this.props.navigation.addListener('willFocus', () => {
      this.props.togglePrimaryColor(PRIMARY_COLORS.BLUE.name);
    });
  }

  render() {
    const { theme } = this.props.screenProps;
    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <Text> Decoding </Text>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  togglePrimaryColor: color => dispatch(togglePrimaryColor(color)),
});

export default connect(null, mapDispatchToProps)(Decoding);
