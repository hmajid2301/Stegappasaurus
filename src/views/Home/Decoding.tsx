import {
  createStackNavigator,
  NavigationRoute,
  NavigationScreenProp
} from "react-navigation";

import Main from "./Decoding/Main";
import Message from "./Decoding/Message";
import Progress from "./Decoding/Progress";

const DecodeNavigator = createStackNavigator({
  Main: {
    navigationOptions: {
      header: null
    },
    screen: Main
  },

  DecodingMessage: {
    navigationOptions: {
      header: null,
      tabBarVisible: false
    },
    screen: Message
  },

  DecodingProgress: {
    navigationOptions: {
      header: null,
      tabBarVisible: false
    },
    screen: Progress
  }
});

DecodeNavigator.navigationOptions = ({
  navigation
}: {
  navigation: NavigationScreenProp<NavigationRoute>;
}) => ({
  swipeEnabled: navigation.state.index === 0,
  tabBarVisible: navigation.state.index === 0
});

export default DecodeNavigator;
