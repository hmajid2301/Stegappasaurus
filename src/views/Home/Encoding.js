import { createStackNavigator } from 'react-navigation';

import EncodeImage from './Encoding/EncodeImage';
import Main from './Encoding/Main';


const EncodeNavigator = createStackNavigator({
  Main: {
    screen: Main,
    navigationOptions: {
      header: null,
    },
  },

  EncodeImage: {
    screen: EncodeImage,
    navigationOptions: {
      header: null,
      tabBarVisible: false,
    },
  },
});

EncodeNavigator.navigationOptions = ({ navigation }) => ({
  tabBarVisible: navigation.state.index === 0,
  swipeEnabled: navigation.state.index === 0,
});

export default EncodeNavigator;
