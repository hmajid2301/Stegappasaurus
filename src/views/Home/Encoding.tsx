import {
  createStackNavigator,
  NavigationRoute,
  NavigationScreenProp
} from "react-navigation";

import Main from "./Encoding/Main";
import Message from "./Encoding/Message";
import Progress from "./Encoding/Progress";

const EncodeNavigator = createStackNavigator({
  Main: {
    navigationOptions: {
      header: null
    },
    screen: Main
  },
  Message: {
    navigationOptions: {
      header: null,
      tabBarVisible: false
    },
    screen: Message
  },
  Progress: {
    navigationOptions: {
      header: null,
      tabBarVisible: false
    },
    screen: Progress
  }
});

EncodeNavigator.navigationOptions = ({
  navigation
}: {
  navigation: NavigationScreenProp<NavigationRoute>;
}) => ({
  swipeEnabled: navigation.state.index === 0,
  tabBarVisible: navigation.state.index === 0
});

export default EncodeNavigator;
