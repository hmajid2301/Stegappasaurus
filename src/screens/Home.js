import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import { createMaterialTopTabNavigator } from 'react-navigation';
import { Icon } from 'native-base';
import { connect } from 'react-redux';

import Header from '../components/Header';
import { colors, fonts } from '../util/styles';
import Encoding from './Encoding';
import Decoding from './Decoding';


const commonTabOptions = color => ({
  activeTintColor: colors.pureWhite,
  pressColor: colors.pureWhite,
  inactiveTintColor: '#ddd',
  labelStyle: {
    fontFamily: fonts.body_xl,
    fontSize: 12,
  },
  indicatorStyle: {
    backgroundColor: colors.pureWhite,
  },
  style: {
    backgroundColor: color,
  },
});

const CustomTabNavigator = createMaterialTopTabNavigator({
  Encoding: {
    screen: Encoding,
    navigationOptions: {
      tabBarLabel: 'Encoding',
      tabBarOptions: commonTabOptions(colors.primary),
    },
  },
  Decoding: {
    screen: Decoding,
    navigationOptions: {
      tabBarLabel: 'Decoding',
      tabBarOptions: commonTabOptions(colors.secondary),
    },
  },
}, {
  tabBarPosition: 'bottom',
});

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

class Home extends Component {
  static navigationOptions = {
    drawerLabel: 'Home',
    drawerIcon: () => (
      <Icon name='home' type='FontAwesome'/>
    ),
  };

  static propTypes = {
    navigation: PropTypes.object.isRequired,
    color: PropTypes.object.isRequired,
  };

  static router = CustomTabNavigator.router;

  render() {
    return (
      <View style={styles.container}>
        <Header color={this.props.color.hexCode} navigation={this.props.navigation} />
        <CustomTabNavigator navigation={this.props.navigation}/>
      </View>
    );
  }
}


const mapStateToProps = state => ({
  color: state.ToggleTheme.colorData,
});

export default connect(mapStateToProps, null)(Home);
