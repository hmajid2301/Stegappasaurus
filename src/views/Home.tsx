import * as React from "react";
import { View } from "react-native";
import {
  createMaterialTopTabNavigator,
  NavigationScreenProp
} from "react-navigation";
import { connect } from "react-redux";

import { ITheme, PrimaryColor } from "@types";
import AppHeader from "~/components/AppHeader";
import { colors, fonts } from "~/modules";
import { IReducerState } from "~/redux/reducers/TogglePrimaryColor";
import Decoding from "~/views/Home/Decoding";
import Encoding from "~/views/Home/Encoding";

import styles from "./Home/styles";

const commonTabOptions = (primaryColor: string) => ({
  activeTintColor: colors.pureWhite,
  inactiveTintColor: "#DDD",
  indicatorStyle: {
    backgroundColor: colors.pureWhite
  },
  labelStyle: {
    fontFamily: fonts.body,
    fontSize: 12
  },
  pressColor: colors.pureWhite,
  style: {
    backgroundColor: primaryColor
  }
});

const TabNavigator = createMaterialTopTabNavigator(
  {
    Decoding: {
      navigationOptions: {
        tabBarLabel: "Decoding",
        tabBarOptions: commonTabOptions(colors.secondary)
      },
      screen: Decoding
    },
    Encoding: {
      navigationOptions: {
        tabBarLabel: "Encoding",
        tabBarOptions: commonTabOptions(colors.primary)
      },
      screen: Encoding
    }
  },
  {
    initialRouteName: "Encoding",
    order: ["Encoding", "Decoding"],
    tabBarPosition: "bottom"
  }
);

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  screenProps: {
    theme: ITheme;
  };
  primaryColor: PrimaryColor;
}

export class Home extends React.Component<IProps, {}> {
  public static router = TabNavigator.router;

  public render() {
    const { theme } = this.props.screenProps;

    return (
      <View style={styles.container}>
        <AppHeader
          primaryColor={this.props.primaryColor}
          navigation={this.props.navigation}
          theme={theme}
        />
        <TabNavigator
          navigation={this.props.navigation}
          screenProps={{ theme }}
        />
      </View>
    );
  }
}

const mapStateToProps = (state: IReducerState) => ({
  primaryColor: state.TogglePrimaryColor.colorData.color
});

export default connect(
  mapStateToProps,
  null
)(Home);
