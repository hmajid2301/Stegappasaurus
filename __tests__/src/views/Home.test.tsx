import { shallow } from "enzyme";
import React from "react";

import Home from "~/views/Home";
describe("Home: Match Snapshots", () => {
  test("1", () => {
    const component = shallow(
      <Home
        navigation={jest.fn() as any}
        screenProps={{
          theme: {
            color: "#17212D",
            background: "#FFF",
            isDark: false
          }
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test("2", () => {
    const component = shallow(
      <Home
        navigation={jest.fn() as any}
        screenProps={{
          theme: {
            color: "#FFF",
            background: "#17212D",
            isDark: false
          }
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
