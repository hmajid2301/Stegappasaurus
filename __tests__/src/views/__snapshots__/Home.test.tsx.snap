// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home: Match Snapshots 1 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <AppHeader
    navigation={[MockFunction]}
    primaryColor="#009CFF"
    theme={
      Object {
        "background": "#FFF",
        "color": "#17212D",
        "isDark": false,
      }
    }
  />
  <Navigator
    navigation={[MockFunction]}
    screenProps={
      Object {
        "theme": Object {
          "background": "#FFF",
          "color": "#17212D",
          "isDark": false,
        },
      }
    }
  />
</View>
`;

exports[`Home: Match Snapshots 2 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <AppHeader
    navigation={[MockFunction]}
    primaryColor="#009CFF"
    theme={
      Object {
        "background": "#17212D",
        "color": "#FFF",
        "isDark": false,
      }
    }
  />
  <Navigator
    navigation={[MockFunction]}
    screenProps={
      Object {
        "theme": Object {
          "background": "#17212D",
          "color": "#FFF",
          "isDark": false,
        },
      }
    }
  />
</View>
`;
